tests/bash/comp-tests.bash

```diff
@@ -154,6 +154,26 @@ _completionTests_verifyCompletion "testprog --customComp f" "firstComp forthComp
 _completionTests_verifyCompletion "testprog --customComp=" "firstComp secondComp forthComp" nofile
 _completionTests_verifyCompletion "testprog --customComp=f" "firstComp forthComp" nofile
 
+#################################################
+# Special characters
+#################################################
+if [ "$BASHCOMP_VERSION" = bash2 ]; then
+    orig_bash_comp_no_sort="${BASH_COMP_NO_SORT}"
+    BASH_COMP_NO_SORT=1
+    _completionTests_verifyCompletion "testprog prefix special-chars bash" "bash space bash\\escape bash\\ escaped\\ space bash>redirect bash#comment bash\$var bash|pipe bash;semicolon"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash\\e" "bash\\escape"
+    # TODO: completionTests_verifyCompletion doesn't support testing something
+    # like this.
+    #_completionTests_verifyCompletion "testprog prefix special-chars bash\\ e" "bash\\ escaped\\ space"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash>r" "bash>redirect"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash#c" "bash#comment"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash\$v" 'bash$var'
+    _completionTests_verifyCompletion "testprog prefix special-chars bash|p" "bash|pipe"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash;s" "bash;semicolon"
```

Comment:
This does not seem right as a result for the completion.
The idea is that if the completion choice is `bash>redirect` then it must appear on the shell as `bash\>redirect`. 

Shell completions should not insert special shell characters but should instead escape them.
So, when doing `testprog prefix special-chars bash|p[tab][tab]`, we should get `testprog prefix special-chars bash\|pipe`.
I'm not sure that is even possible to achieve with bash, but at least, if there is only one completion choice and it is  `bash|pipe`, then `testprog prefix special-chars ba[tab][tab]` should complete to `testprog prefix special-chars bash|pipe`
Same goes for the other special chars except `#`, which seems to be passed to the program without needing to be espcaped.

A good way to see how the completions should behave is to compare them with zsh.
```
zsh
$ source <(testprog completion zsh)
$ testprog prefix special-chars bash4<TAB>

# becomes
$ testprog prefix special-chars bash4\>redirect
```

Reply:
Done